torch/utils/_benchmark/utils/valgrind_wrapper/timer_interface.py

```diff
@@ -0,0 +1,448 @@
+"""Intermediate layer between `Timer` and `valgrind`."""
+import collections
+import dataclasses
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import DefaultDict, Dict, Optional, Tuple
+
+import torch
+
+
+@dataclasses.dataclass(repr=False, eq=False, frozen=True)
+class CallgrindStats(object):
+    stmt: str
+    setup: str
+    number: int
+    num_threads: int
+    built_with_debug_symbols: bool
+    baseline_inclusive_stats: Tuple[Tuple[int, str], ...]
```

Comment:
Consider defining a named tuple for these `Tuple[int, str]` here

Reply:
Done. I agree that is much nicer.